components/Forms/InteractionForm/InteractionForm.js

```diff
@@ -0,0 +1,150 @@
+import InteractionDetailsHeader from "../../InteractionDetailsHeader/InteractionDetailsHeader";
+import {
+  ButtonsContainer,
+  Heading,
+  StickyContainer,
+  StyledFieldset,
+  StyledTextarea,
+} from "./InteractionForm.styled";
+import { useState } from "react";
+import SingleLineInput from "../SingleLineInput/SingleLineInput";
+import { baseInteractionTypes } from "@/data/BaseData";
+import { getFullName, getFullSortName } from "@/utils/getContactDetails";
+import Button from "@/components/Button/Button";
+import { useRouter } from "next/router";
+
+export default function InteractionForm({
+  onSubmitForm,
+  interaction,
+  contacts,
+}) {
+  const router = useRouter();
+
+  const [currentInteraction, setCurrentInteraction] = useState(interaction);
+  const isEqual =
+    JSON.stringify(currentInteraction) === JSON.stringify(interaction);
+
+  const sortedContacts = contacts.slice().sort((a, b) => {
+    const nameA = getFullSortName(a);
+    const nameB = getFullSortName(b);
+    if (nameA < nameB) return -1;
+    if (nameA > nameB) return 1;
+    return 0;
+  });
+
+  const participantOptions = sortedContacts.map((contact) => {
+    return { value: contact.id, label: getFullName(contact) };
+  });
+
+  function handleUserInput(event, fieldName) {
+    if (fieldName === "interactionType") {
+      setCurrentInteraction({
+        ...currentInteraction,
+        type: event.value,
+      });
+    } else if (fieldName === "participants") {
+      const participantsArray = event.map((participant) => participant.value);
+      setCurrentInteraction({
+        ...currentInteraction,
+        participants: participantsArray,
+      });
+    } else {
+      setCurrentInteraction({
+        ...currentInteraction,
+        [fieldName]: event.target.value,
+      });
+    }
+  }
+
+  const selectableSortedInteractionTypes = baseInteractionTypes
+    .filter((baseInteractionType) => baseInteractionType.isSelectable)
+    .sort((a, b) => {
+      const typeA = a.type.toUpperCase();
+      const typeB = b.type.toUpperCase();
+      if (typeA < typeB) {
+        return -1;
+      }
+      if (typeA > typeB) {
+        return 1;
+      }
+      return 0;
+    })
+    .map((baseInteractionType) => {
+      if (baseInteractionType.isSelectable) {
```

Comment:
Why do you need this if statement? You have already applied the filter before.`.filter((baseInteractionType) => baseInteractionType.isSelectable)`

Reply:
The map array method does not work without if on strings.